pkg/apply/apply.go

```diff
@@ -30,49 +32,35 @@ func ApplyObject(ctx context.Context, client k8sclient.Client, obj *uns.Unstruct
 	objDesc := fmt.Sprintf("(%s) %s/%s", gvk.String(), namespace, name)
 	log.Printf("reconciling %s", objDesc)
 
-	if err := IsObjectSupported(obj); err != nil {
-		return errors.Wrapf(err, "object %s unsupported", objDesc)
+	// determine resource
+	rm, err := client.RESTMapper().RESTMapping(gvk.GroupKind(), gvk.Version)
+	if err != nil {
+		return fmt.Errorf("failed to retrieve resource from Object %s", objDesc)
 	}
 
-	// Get existing
-	existing := &uns.Unstructured{}
-	existing.SetGroupVersionKind(gvk)
-	err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
-		err := client.Get(ctx, types.NamespacedName{Name: obj.GetName(), Namespace: obj.GetNamespace()}, existing)
+	fieldManager := "cluster-network-operator"
+	if name, ok := ctx.Value(names.KEY_CONTROLLER_NAME).(string); ok {
+		fieldManager = fmt.Sprintf("%s/%s", fieldManager, name)
+	}
 
-		if err != nil && apierrors.IsNotFound(err) {
-			log.Printf("does not exist, creating %s", objDesc)
-			err := client.Create(ctx, obj)
-			if err != nil {
-				log.Printf("create of %s was unsucessful", objDesc)
-				return err
-			}
-			log.Printf("successfully created %s", objDesc)
-			return nil
+	// Use server-side apply to merge the desired object with the object on disk
+	err = retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		t := true
+		patchOptions := metav1.PatchOptions{
+			// It is considered best-practice for controllers to force
+			Force:        &t,
+			FieldManager: fieldManager,
 		}
+		// Send the full object to be applied on the server side.
+		data, err := runtime.Encode(unstructured.UnstructuredJSONScheme, obj)
 		if err != nil {
-			log.Printf("could not retrieve %s", objDesc)
-			return err
-		}
-
-		// object exists - for create-only objects, stop here
-		if anno := existing.GetAnnotations()[names.CreateOnlyAnnotation]; anno == "true" {
```

Comment:
Oops, I realize I forgot this.

Reply:
On second thought, we don't need this anymore -- since we now merge fields, it's no longer a problem.